src/Hubspot/SyncToService.php

```diff
@@ -103,54 +119,175 @@ public function sync(EntityInterface $entity) {
     if (empty($field_mapping)) {
       return;
     }
+
+    // Prepare the paylaod to send to Hubspot.
+    $hubspot_field_properties = $this->preparePayload($field_mapping);
+    if (empty($hubspot_field_properties)) {
+      return;
+    }
+
+    // Now, do the actual syncing depending on the entity type.
+    $function_name = 'sync' . $entity_mapping['type'];
+    return $this->$function_name($hubspot_field_properties, $entity_mapping['id']);
+  }
+
+  /**
+   * Prepare the payload for syncing the properties.
+   *
+   * @param array $field_mapping
+   *   The array of fields that should be mapped.
+   *
+   * @return mixed
+   *   An array of Hubspot properties with their values.
+   */
+  protected function preparePayload(array $field_mapping) {
+    $hubspot_field_properties = [];
+    foreach ($field_mapping as $drupal_field_name => $hubspot_field) {
+      if (!$hubspot_field['status']) {
+        return;
+      }
+
+      $hubspot_field_properties[] = [
+        'property' => $hubspot_field['id'],
+        'value' => $hubspot_field['value'],
+      ];
+    }
```

Comment:
I'm wondering, as we discussed HubSpot requires two identifiers when you create a field: a property and a hubspot field ID. So, here we would be passing the property, which would not necessarily be the hubspot field ID but a unique identifier of the field on our side? And then HubSpot would know from the field mapping that this correspond to that hubspot field ID.

Reply:
You're right. Once, I'm able to successfully do the property mapping to enable the eCommerce bridge, we can confirm the assumption.